### ISSUE TITLE
Unable to use the --spacing() function from tailwind V4

### ISSUE THREAD

**Issue by The-Code-Monkey** [schoero/eslint-plugin-readable-tailwind#97]
The --spacing(...) function requires that the `--spacing` theme variable exists, but it was not found.

This only happens after giving the plugin the entryPoint to my globals.scss file before that it was fine.

**Comment by schoero** [comment]
> The --spacing(...) function requires that the --spacing theme variable exists, but it was not found.

Please elaborate on this a little more.
Is this an error that the ESLint plugin is throwing? Does the spacing actually work in your HTML? 

<br />

The tailwindcss `--spacing()` function  [relies on the `--spacing` variable](https://tailwindcss.com/docs/functions-and-directives#spacing-function) to work. The `--spacing` variable usually comes from the default tailwind theme. 

Make sure that you importing `tailwindcss` correctly in your `global.scss`:

```css
/* global.scss */
@import "tailwindcss"; 
```

If you have imported tailwindcss correctly, make sure that you **don't** [reset the theme](https://tailwindcss.com/docs/theme#using-a-custom-theme) or the spacings in your `global.scss` file.

```css
@theme {
  /* resets the tailwindcss default theme */
  --*: initial;

  /* resets the spacings */
  --spacing-*: initial;
}
```

If you actually want to change the spacings, you need to set the `--spacing` variable in your theme.

**Comment by The-Code-Monkey** [comment]
Hi, yeah it's the sort classes rule that's failing from your package with that error, 

I have tailwind imported at the top and I have not overridden the --spacing variable, could it be that I'm importing tailwind with @use instead of @import

**Comment by schoero** [comment]
Yes it could be. Can you try if it works with `@import` or is there a specific reason why you want to use `@use` and not `@import`?

**Comment by The-Code-Monkey** [comment]
Hi Yeah I've checked and it seems to be because im using @use instead of @import is import is not the preferred way to include other sass files according to sass 

https://sass-lang.com/documentation/at-rules/use/#differences-from-import
https://medium.com/@philip.mutua/difference-between-use-and-import-in-scss-1cb6f501e649

**Comment by The-Code-Monkey** [comment]
im happy to use import for now but if use is the preferred way i would like to be able to switch to that in the future.

**Comment by schoero** [comment]
tailwindcss recommends to not use a preprocessor like sass: https://tailwindcss.com/docs/compatibility#sass-less-and-stylus

I think tailwindcss [handles `@import` internally at build time](https://tailwindcss.com/docs/compatibility#build-time-imports) but they don't handle `@use`.

If you want to stick with Sass and `@use` then Sass itself would need to handle the injection of tailwindcss with `@use "tailwindcss"` correctly.

**Comment by The-Code-Monkey** [comment]
That is a good point i didnt even know that. Thanks for letting me know. ill remove the use of sass.
